### PR TITLE
Assignment

### PR DESCRIPTION
**Pull request by wma5283**
- pdf upload
- pdf download

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,8 +1,9 @@
 {
-  "homepage": "http://AlanWu886.github.io/react_sample",
+  "homepage": "http://localhost:3000/",
   "name": "react_sample",
   "version": "0.1.0",
   "private": true,
+  "proxy": "http://cors-anywhere.myapi.herokuapp.com",
```
Comment (wma5283) [inline package.json:6]: Do you still need proxy

Comment (wma5283) [reply package.json:6]: You are not supposed to use the cors-anywhere url as a proxy here I don't think. My understanding is you need to append this URL as a prefix to your original URL

File: src/client/App.js [hunk 2]
```diff
@@ -32,6 +32,11 @@ class App extends React.Component {
       this.setState({
         isLoading: false
       })
+
+      // Simple GET request using fetch
```
Comment (wma5283) [inline src/client/App.js:36]: delete this comment?

File: src/client/Products.js [hunk 3]
```diff
@@ -27,6 +27,16 @@ class Products extends React.Component {
     this.toggleDisplay = this.toggleDisplay.bind(this)
   }
 
+  componentDidMount() {
+   const apiUrl = 'https://codetoad.herokuapp.com/v1/codetoad/course/details/all';
```
Comment (wma5283) [inline src/client/Products.js:31]: Why fetch here? Why not in App.js?

File: src/client/components/Assignment.js [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+  handleAnswer(event) {
+    console.log(event.target.value);
+    console.log(event.target.id);
+    let index = event.target.id
+    let answers = [...this.state.answers];
+    answers[index] = event.target.value;
+    this.setState({answers:answers})
+    // 4. Put it back int
+    console.log(index);
+    // var item = {...this.prop.item}
```
Comment (wma5283) [inline src/client/components/Assignment.js:46]: Delete these comments. It can add confusion later on

File: src/client/components/Assignments.js [hunk 5]
```diff
@@ -1,19 +1,78 @@
...
+      <h1>No assignment at this momoent</h1>
         return(
             <div>
                 <div>
-                    Assignments page
+                  <Row>
+                    {assignmentList}
+                  </Row>
+                    <input type="file" name="file" onChange={(e) => this.onChange(e)} />
+                    <PdfGenerator></PdfGenerator>
```
Comment (wma5283) [inline src/client/components/Assignments.js:55]: I think this can simply be <PdfGenerator/>

File: src/reducer.js [hunk 6]
```diff
@@ -17,6 +16,16 @@ const initState = {
       "description": "Description, properties, and storage allocation functions of data structures including balanced binary trees, B-Trees, hash tables, skiplists, tries, KD-Trees and Quadtrees. Algorithms for manipulating structures. Applications from areas such as String Processing, Computer Graphics, Information Retrieval, Computer Networks, Computer Vision, and Operating Systems.",
       "price": "$10.00",
     }
+  ],
+  catalog:[],
```
Comment (wma5283) [inline src/reducer.js:20]: I think we want to do some very general API GET requests in the reducer

File: src/client/components/Assignments.js (hunk 5 above)
Comment (AlanWu886) [reply src/client/components/Assignments.js:55]: you mean the title?

File: src/reducer.js (hunk 6 above)
Comment (AlanWu886) [reply src/reducer.js:20]: yep

**Pull request merged**